src/sonic-build-hooks/hooks/git

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+parse_config(){
+    . /usr/local/share/buildinfo/scripts/buildinfo_base.sh
+    REAL_COMMAND=$(get_command git)
+
+    version_file=$VERSION_PATH/versions-git
+    new_version_file=$BUILD_VERSION_PATH/versions-git
+
+    MODE_CLONE=0
+    # parse input parameters
+    for i in "$@"
+    do
+        if [[ $i == "clone" ]];then
+            MODE_CLONE=1
+        fi
+    done
+}
+
+get_clone_path(){
+    # get paremater of clone path
+    while (( "$#" )); do
+        case $1 in
+            -b|--branch|--reference|--reference-if-able|-c|--config|--origin|-u|--upload-pack|-j|--jobs|--depth|--dissociate)
+                shift 2
+                ;;
+            clone|-l|--local|--no-hardlinks|-s|--shared|--dissociate|-q|--quiet|-v|--verbose|--progress|--server-option=*|--bare|--sparse|--filter=*|--template=*|--mirror|--reference|--shallow-*|--no-tags|--recurse-submodules*|--remote-submodules|--no-remote-submodules|--separate-git-dir*)
+                shift 1
+                ;;
+            *)
+                if [[ $URL == "" ]];then
```

Comment:
Looks like you try to exclude all the known parameters, then do an assumption, the value is URL, is it true?
How about using the web pattern? like git: http: https, something like it.
How about excluding all the parameter starts with "-"

Reply:
no. we need to distinguish 2 scenario:
1. --depth 2
2. --quiet